### PR TITLE
sartre: new package

### PR DESCRIPTION
**Pull request by wdconinc**
[Sartre](https://sartre.hepforge.org) is an event generator for exclusive diffractive vector meson production and DVCS in ep and eA collisions based on the dipole model.

Notable dependencies are [`cuba`](https://github.com/spack/spack/pull/32510).

TODO:
- [ ] ~~This version is still forcing the use of the vendored package. Discussion is ongoing with the developer.~~

### CODE REVIEW

**Comment by alalazo** [comment]
What is the status of this PR?

**Comment by wdconinc** [comment]
> Discussion is ongoing with the developer.

This is still not resolved (not gotten an answer). I'll wrap this up today without developer input. I'll just live with the vendored dependency or filter_file it out.

File: var/spack/repos/builtin/packages/sartre/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+class Sartre(CMakePackage):
+    """Sartre 1 is an event generator for exclusive diffractive vector
+    meson production and DVCS in ep and eA collisions based on the
+    dipole model."""
+
+    homepage = "https://sartre.hepforge.org"
+    url = "https://sartre.hepforge.org/downloads/?f=sartre-1.39-src.tgz"
+    list_url = "https://sartre.hepforge.org/downloads/"
+
+    maintainers = ["wdconinc"]
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/sartre/package.py]:
```suggestion
    maintainers("wdconinc")
```

File: var/spack/repos/builtin/packages/sartre/package.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/sartre/package.py]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

**Comment by wdconinc** [comment]
Things I would have wished the developers could address (but they haven't):
- `/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.apple.lastuseddate#PS'` (stuff included in tar archive that is not really supposed to be there; though maybe spack could be less loud about this too, `tar --warning=no-unknown-keyword` hides this),
- `cuba` is forcibly used as vendored dependency in a way that's hard to simply `filter_file` out,
- the `CMakeLists.txt` installs in `<prefix>/sartre`, which I did `filter_file` out.

**Comment by wdconinc** [comment]
@spackbot fix style

**Review by alalazo — APPROVED** [review]

**Pull request merged**